### PR TITLE
Adding application insights support to REST app and also update documentation

### PR DESCRIPTION
**Pull request by jananivMS**

### CODE REVIEW

**Comment by WilliamMortlMicrosoft** [comment]
at a quick glance - this looks correct, but I haven't run it locally and am not an expert on typescript on this so I'll hold off approving until a more knowledgeable person can

File: src/app/controllers/movie.ts [hunk 1]
```diff
@@ -1,26 +1,30 @@
+import {telemetryClient} from '../../server'
 
-/**
- *  Retrieve and return all movies 
- *  Filter movies by name "?q=<name>"
+/** Retrieve and return all movies from the database.
```
Comment (aka-msft) [inline src/app/controllers/movie.ts]: I think you have an old version of the docstring here.  Can you try to do a "git pull" from helium master and push to your PR again?

File: src/app/controllers/movie.ts [hunk 2]
```diff
@@ -1,26 +1,30 @@
...
 export function getAll(req, res) {
     // Stub:
+    telemetryClient.trackEvent({name: "getAll endpoint"})
     return res.send(200, {message: 'Successfully reached getAll endpoint.'});
 };
 
-/**
- *  Create a movie
- */
+// Create a movie
```
Comment (aka-msft) [inline src/app/controllers/movie.ts]: Ditto - old docstring, Erin updated them in one of his PR's yesterday

File: src/app/routes/movie.ts [hunk 3]
```diff
@@ -3,11 +3,18 @@ import * as movies from '../controllers/movie'
 /**
  * Movie endpoint routes
  */
-export function registerRoutes (app) {
+
+export function registerRoutes (app, telemetryClient) {
```
Comment (aka-msft) [inline src/app/routes/movie.ts]: Ok good, glad to see this did work

File: src/app/routes/movie.ts [hunk 4]
```diff
@@ -3,11 +3,18 @@ import * as movies from '../controllers/movie'
...
 
     // @todo move up a level so other entities can inherit
     const apiPrefix = "/api";
 
+    // Health check - pings CosmosDB
```
Comment (aka-msft) [inline src/app/routes/movie.ts]: This section is also old - Erin removed this and implemented the health check yesterday.  "git pull" should allow you to fix

File: src/server.ts [hunk 5]
```diff
@@ -19,10 +38,12 @@ server.get('/', (req, res) => {
     res.json({"message": "Welcome to the MovieInfo reference application."});
 });
 
-systemRoutes.registerRoutes(server);
-movieRoutes.registerRoutes(server);
+telemetryClient.trackEvent({name: "Registering routes"}); 
+systemRoutes.registerRoutes(server, telemetryClient);
```
Comment (aka-msft) [inline src/server.ts]: Looks good to me, hoping this works!

File: src/app/controllers/movie.ts (hunk 1 above)
Comment (jananivMS) [reply src/app/controllers/movie.ts]: I jsut did a git pull from upstream master and fixed the merge conflicts

File: src/app/controllers/movie.ts (hunk 2 above)
Comment (jananivMS) [reply src/app/controllers/movie.ts]: I think it updated it. It also came in with Sean;s changes. So please take a look to see if it looks ok now?

File: src/app/controllers/movie.ts (hunk 1 above)
Comment (aka-msft) [reply src/app/controllers/movie.ts]: Still old docstring here
New one reads as:

/**
 *  Retrieve and return all movies 
 *  Filter movies by name "?q=<name>"
 */

File: src/app/routes/movie.ts (hunk 4 above)
Comment (aka-msft) [reply src/app/routes/movie.ts]: The `// Health check` commented section in routes/movie.ts is old

**Comment by seushermsft** [comment]
> Looks good.
> 
> Amanda made some good comments that should probably be addressed wrt to code regressions.
> 
> Do we need a follow-up issue created to track what needs to be done after the CosmosDB work lands?

I filed #47 before merging the CosmosDB changes to add logging in metrics.

**Comment by jananivMS** [comment]
For testing this code, the environment variable APPINSIGHTS_INSTRUMENTATIONKEY needs to be set to the instrumentation key of an Application Insights instance.

File: docs/azure-infrastructure.md [hunk 6]
```diff
@@ -4,7 +4,11 @@
 
 Before getting started, make sure to have the [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest) installed on your machine. 
 
-## Decide on an Application Prefix
+You should also have an active Azure subscription.
+
+## Decide on an Application Prefix and Resource Group Prefix
+
+If you have multiple people working off the same Azure subscription, decide on a prefix to make the resource group name unique. For instance, "ABC_Helium" would be a good resource group name where "ABC" is a unique prefix (your alias/initials would be a good one!). Throughout this documentation, where you see **{resource_group_name}** you should replace that with your unique resource group name
```
Comment (WilliamMortlMicrosoft) [inline docs/azure-infrastructure.md]: these two paragraphs should be unified together and {app prefix} should just be appended to the resource group name... so something like: {app prefix}helium

**Comment by timfpark** [comment]
In the fullness of time (eg. after the first hack), we will also want to generalize metrics monitoring to enable both an App Insights model and/or a Prometheus model based on the configuration passed to the application since our customers will want references with both.  This implies that perhaps we should centralize the metrics monitoring logic in a service such that we can put the logic on which (or both) to use in one place.

**Pull request merged**